internal/security/setup/option/optionHelpers.go

```diff
@@ -214,20 +233,64 @@ func markComplete(dirPath string) error {
 	return nil
 }
 
-func getXdgRuntimeDir() string {
+func getWorkDir() string {
+	var workDir string
+	var err error
+
 	if xdgRuntimeDir, ok := os.LookupEnv(envXdgRuntimeDir); ok {
-		return xdgRuntimeDir
+		workDir = filepath.Join(xdgRuntimeDir, pkiInitBaseDir)
+	} else if setup.Configuration.SecretsSetup.WorkDir != "" {
+		workDir, err = filepath.Abs(setup.Configuration.SecretsSetup.WorkDir)
+		if err != nil {
+			// use default work dir if error occurs
+			workDir = filepath.Join(defaultWorkDir, pkiInitBaseDir)
+		}
+	} else {
+		// if not specified in config and $XDG_RUNTIME_DIR is undefined, default to /tmp
+		workDir = filepath.Join(defaultWorkDir, pkiInitBaseDir)
+	}
+
+	return workDir
+}
+
+func getCertConfigDir() (string, error) {
+	var certConfigDir string
+	if setup.Configuration.SecretsSetup.CertConfigDir != "" {
+		certConfigDir = setup.Configuration.SecretsSetup.CertConfigDir
+		if exist, _ := checkDirExistsAndIsWritable(certConfigDir); !exist {
```

Comment:
a comment here explaining why we don't care if it's writable would be appreciated